pkgs/tools/misc/diffoscope/default.nix

```diff
@@ -3,7 +3,8 @@
 , e2fsprogs, enjarify, file, findutils, fontforge-fonttools, ffmpeg, fpc, gettext, ghc, ghostscriptX, giflib, gnumeric, gnupg, gnutar
 , gzip, html2text, hdf5, imagemagick, jdk, libarchive, libcaca, llvm, lz4, mono, ocaml, oggvideotools, openssh, openssl, pdftk, pgpdump, poppler_utils, procyon, qemu, R
 , radare2, sng, sqlite, squashfsTools, tcpdump, ubootTools, odt2txt, unzip, wabt, xmlbeans, xxd, xz, zip, zstd
-, enableBloat ? false
+, enableBloat ? !stdenv.isDarwin
+, enableUnfree ? false
```

Comment:
there probably is no policy, but shouldn't this default to `nixpkgs.config.allowUnfree`?

and in a good UX, shouldn't it also work when allowing the specific dependency with `allowUnfreePredicate` or `allowlistedLicenses` etc?

maybe this needs more design and a proper implementation and policy for nixpkgs

Reply:
it would create the surprising behavior where depending on allowUnfree or not, you would have to compile or have a different package from the named one

Reply:
>you would have to compile

yes, i thought about that

but documentation already sais:

>Note that we are not able to test or build unfree software on Hydra due to policy.

source: https://nixos.org/manual/nixpkgs/stable/#chap-packageconfig

so shouldn't surprise

>have a different package from the named one

you mean when one person with `nixpkgs.config.allowUnfree = true;` installs diffoscope, they have a different package than one person with `false`?

it's kind of implicit, but i can see that as expected behavior

it would also be different when you override it explicitly

i think this is ok for now, but it's an interesting topic for the Nixpkgs Architecture Team to create a proper interface for packages

Reply:
> it's kind of implicit, but i can see that as expected behavior

Beyond "expected" or not, it makes it impossible to **not rebuild** diffoscope if you have allowUnfree without overriding the package, which is, IMHO, very bad UX.

A user probably want the cache, otherwise you will plenty of people saying "why am I compiling this" (unfree=true) while people will say "huh no it works with me" (unfree=false).

So it is definitely surprising to my eyes.

Reply:
> there probably is no policy, but shouldn't this default to `nixpkgs.config.allowUnfree`?

As far as I can tell, we 100% consistently do not do this in Nixpkgs.